### ISSUE TITLE
gpgme v 1.10.0 KDL's

### ISSUE THREAD

**Issue by Begasus** [haikuports/haikuports#2040]
When trying to update gpgme to the latest version I ran into KDL everytime (on clean build)
Adding links to the builds, maybe someone can have a look at it.

New version (with sc in KDL) (serial output) (adjusted recipe):
https://pastebin.com/h7jsdYX3

Old version 1.9.0 still ok (serial output):
https://pastebin.com/RRnsShDJ

New version 1.10.0, recipe duplicated from 1.9.0 (serial output):
https://pastebin.com/xaSBFDta

Running "co" in KDL did build the recipe but in the end after "grabbing" it didn't "unmount" the filesystem making the need of a hard reboot ...

**Comment by mmlr** [comment]
KDLs generally always need to be tracked at https://dev.haiku-os.org/ as they are Haiku bugs. That said, I am able to reproduce this and a first look indicates that the team tries to wait on itself as a child of itself, which locks the team and then tries to lock the team again as a child. Apparently no validation of the parent/child relationship is being done.

**Comment by mmlr** [comment]
The double lock is fixed in hrev51725.

Unmounting still fails because the tests run during the build start a gpg-agent that isn't cleaned up and then causes the volume to be busy. The test indicates that it failed to find the `which` command because it isn't in the chroot. Adding `cmd:which` to `BUILD_PREREQUIRES` may or may not affect that.

**Comment by Begasus** [comment]
Confirmed, KDL is fixed after the update to latest hrev, unmounting is still there, even with cmd:which added in the recipe, should I still file a ticket for this at the tracker?
https://pastebin.com/ZV57nqWx

**Comment by mmlr** [comment]
You mean the Haiku bugtracker? No, it is not a bug. The build process just starts an application and leaves it running, making the volume busy. Everything that happens then is correct, i.e. the volume not unmounting and what HaikuPorter reports/does with that.

The build process should be fixed to kill the agent it starts, as that is the right thing to do. If it isn't straight forward to do that, disabling tests during build may work. Generally moving the tests to TEST() would be nicer IMO.

**Comment by Begasus** [comment]
Ok, disabling the tests worked, thanks again, this one can be closed as it's not related to haikuporter.

**Comment by Begasus** [comment]
For testing purposes I ran the test (with the tests enabled with configure) and ended up in KDL again, I think we better stick to the last know to work ok version 1.9.0 
https://pastebin.com/S834Ew4r

**Comment by mmlr** [comment]
Please open bug reports at the Haiku bug tracker for any new KDLs you encounter. This one seems good at producing them.

**Comment by Begasus** [comment]
Ticket created ...

**Comment by ttcoder** [comment]
Probably not much use adding on a closed ticket in the wrong place (but then again I have a *policy* against commenting in that other place *g*)..

But anyway FWIW: that latter stack trace (TCP's looper thread) seems similar to the one fixed by pulkomandy here:
https://dev.haiku-os.org/ticket/12567

Except it's in TCP instead of UDP code. For months we've had clients reporting KDLs with that stack trace either in TCP or UDP code, so I wondered why it was fixed only in UDP but not in TCP.. Now I wonder even more :-)

**Comment by pulkomandy** [comment]
This would rather be raised in that particular ticket. I hit this problem several times in UDP code, but don't remember seeing it on the TCP side. I fixed the bug I could see and the ticket did not mention TCP either. So I fixed what I was aware of.

@Begasus: could you please add a link to the ticket here, so we can track the issue at the right place?

**Comment by ttcoder** [comment]
@pulkomandy - Obviously; and gotta thank you for that, because that one was particularly annoying to us; and indeed I don't remember seeing a ticket (on the Haiku trac) about TCP too. I was just wondering why some use-cases (our stations) trigger both KDLs, whereas the rest of the world just saw the UDP KDL, is all. And hoping my pointer could be useful to someone in fixing this (second part of the) bug.

Note that https://dev.haiku-os.org/ticket/13927 does not contain the KDL stack trace; the message above in this treahd does however (though it's hard to read)

**Comment by Begasus** [comment]
I could try a clean build with a clean syslog if needed with a included stack trace and teams, or probably better at the bug tracker?

**Comment by mmlr** [comment]
Please don't start to mix things even more. This issue was about the initial KDL. It was filed in the wrong issue tracker to begin with, is now fixed and should therefore stay closed.

A new issue may be opened about the test gpgme runs during build that isn't cleaned up and then blocks the unmount of the chroot file systems. This issue tracker would be the right place for such an issue.

The TCP KDL is not in any way related to the original KDL described here. The report was moved to the correct bug tracker and should therefore not be commented on here, but at https://dev.haiku-os.org/ticket/13927 instead.
